packages/tao/src/commands/ngcli-adapter.ts

```diff
@@ -472,9 +472,20 @@ export class NxScopeHostUsedForWrappedSchematics extends NxScopedHost {
         return super.read(path);
       }
     } else {
-      // found a matching change in the host
       const match = findMatchingFileChange(this.host, path);
-      return match ? of(Buffer.from(match.content)) : super.read(path);
+      if (match) {
+        // found a matching change in the host
+        return of(Buffer.from(match.content));
+      } else if (
+        // found a change to workspace config, and reading a project config file
+        basename(path) === 'project.json' &&
+        findWorkspaceConfigFileChange(this.host)
+      ) {
+        return of(this.host.read(path));
+      } else {
+        // found neither, use default read method
+        super.read(path);
```

Comment:
@AgentEnder @FrozenPandaz doesn't this code path also need a return statement?

Reply:
Thanks @JamesHenry... It definitely does. I'll put in a cleanup pr and get it merged.